sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/consumer/strategies/MessageListenerStrategy.java

```diff
@@ -143,6 +141,14 @@ private MetadataContextCloseable createMetadataContext(ConsumerRecord<K, V> cons
     return MetadataContext.createCloseableContext(newContext);
   }
 
+  private Headers normalizeHeaders(Headers headers) {
+    return new RecordHeaders(
+        Arrays.stream(headers.toArray())
+            .filter(Objects::nonNull)
+            .map(CaseInsensitiveHeader::new)
+            .collect(Collectors.toList()));
+  }
```

Comment:
Could we go for a simpler approach like the following instead of introducing a new class?

```suggestion
  private Headers normalizeHeaders(Headers headers) {
    return new RecordHeaders(
        Arrays.stream(headers.toArray())
            .filter(Objects::nonNull)
            .map(h -> new RecordHeader(StringUtils.lowerCase(h.key()), h.value()))
            .collect(Collectors.toList()));
  }
```

Reply:
done